src/dbcparser.cpp

```diff
@@ -73,6 +84,159 @@ std::string dos2unix(const std::string& data)
     return noWindowsShit;
 }
 
+#if __cplusplus >= 201402L
+inline auto
+#else
+inline std::map<CANmessage, std::vector<CANsignal>>::iterator
+#endif
+    getMessageIteratorById(CANdb_t &canDb, uint32_t id, bool& found)
+{
+    auto messageIt = std::find_if(canDb.messages.begin(), canDb.messages.end(),
+        [id](const std::pair<const CANmessage, std::vector<CANsignal>>& entry) {
+            return entry.first.id == id;
+        });
+    found = messageIt != canDb.messages.end();
+    return messageIt;
+}
+
+#if __cplusplus >= 201402L
+inline auto
+#else
+inline std::vector<CANsignal>::iterator
+#endif
+    getSignalIteratorByMessageIdAndName(CANdb_t &canDb, uint32_t id,
+        const std::string& name, bool &found)
+{
+    std::vector<CANsignal>::iterator signalIt;
+    auto messageIt = getMessageIteratorById(canDb, id, found);
+    if (found) {
+        signalIt = std::find_if(messageIt->second.begin(),
+            messageIt->second.end(), [name](const CANsignal& signal) {
+                return signal.signal_name == name; });
+        found = signalIt != messageIt->second.end();
+    }
+    return signalIt;
+}
+
+void appendMessageTransmittingEcus(CANdb_t &canDb, uint32_t id,
+    std::deque<std::string>& ecus)
+{
+    cdb_debug("Appending transmitting ECUs for message {}", id);
+    bool messageItFound = false;
+    auto messageIt = getMessageIteratorById(canDb, id, messageItFound);
+    if (messageItFound) {
+        // See notes in setMessageComment regarding this copy/remove/insert
+        // implementation.
+        cdb_debug("Found the message that needs the ECUs");
+        CANmessage updatedMessage(messageIt->first);
+        std::vector<CANsignal> existingSignals(messageIt->second);
+        while (!ecus.empty()) {
```

Comment:
I think set insert/union operations here would make this code/api cleaner, so are we sure we need ecus to be a `std::vector<std::string>` ? or `std::unordered_set<std::string>` should also work?
Alternatively we can just use temporary `std::unordered_set<std::string>` here to do the merge operations.

Reply:
Maybe... but the upstream code works with `std::deque`, also to push and pop when parsing via cpp-peglib, so I really didn't want to fiddle with this too much.